lib/dpl/provider/launchpad.rb

```diff
@@ -0,0 +1,48 @@
+require 'net/http'
+require 'net/https'
+
+module DPL
+  class Provider
+    class Launchpad < Provider
+
+      def initialize(context, options)
+        super
+        @http = Net::HTTP.new('api.launchpad.net', 443)
+        @http.use_ssl = true
+      end
+
+      def check_auth
+      end
+
+      def needs_key?
+        false
+      end
+
+      def push_app
+        response = api_call('/1.0/' + options[:slug] + '/+code-import', {'ws.op' => 'requestImport'})
+        error('Deploy failed! Launchpad credentials invalid. ' + response.code.to_s) if response.code == '401'
+        error('Error: ' + response.code + ' ' + response.body) if response.code != '200'
+      end
+
+      private
+
+        def api_call(path, data)
+          req = Net::HTTP::Post.new(path)
+          req.set_form_data(data)
+          req['Authorization'] = get_authorization_header
+          return @http.request(req)
+        end
+
+        def get_authorization_header
```

Comment:
And to be more nit picking, the value this method returns is not a header, either. Perhaps just `authorization`.